docs/patterns/ci-cd-10000-feet.md

```diff
@@ -0,0 +1,170 @@
+
+So all that we really do here is to fill the CI register with build targets.
+
+However, it is important that Unit Tests are embedded as part of the Nix build pipeline.
+This ensures for everyones sanity, that only a unit-tested build-arifact counts as a successful build.
+
+In the case that we need to ensure sequenciality, the register can be filled in sequential chuncks of targets that can be built in parallel.
+
+### Deployment Phase
+
+```yaml
+Register: __std.ci.deploy # list (order) of attrs (parallel) of deployment actions
+```
+
+The deployment phase schedules deployment artifacts via schematic target state (data-)manifests.
+It pushes the target state onto a (potentially stateful) control loop via a (potentially remote) API interface.
+
+The specific Block Type Action pushed to the `__std.ci.deploy` Register determins _how_ this process is to be accomplished.
+
+> **Push vs Pull Workflows**
+>
+> This concept enacts a push-based workflow as opposed to a pull-based workflow.
+> Push-based workflows are predominant in DevOps, but they actually puncture security holes into your target infrastructure.
+> There is no way around _listening_ on some port for the push to happen.
+>
+> Pull-based workflows don't suffer from this structural shortcoming and we may explore them closer in the future.
+
+> **Orchestration vs Choreography**
+>
+> This concept enacts a form of orchestration.
+> For increasingly complex infrastructure, there is little room for fogoing the benefits of orchestration over choreography, such as having a tigthened control loop.
+>
+> Typically, push-based workflows use orchestration, because the can afford state locally and, hence, cheaply.
+>
+> Typically, pull-based wowkflows use choreography, because they can't orchestrate without global (i.e. remote) state and, hence, complexity.
+>
+> An ideal approach would be an orchestrating pull-based approach via some sort of remote _clearance state machine_ that can be programmed via an API.
+> We might explore this approach in the future.
+
+Sequenciality can be achieved via the Register the same way as in the other phases.
+However, you need to understand, that only a single ("development") environment is ready for automatic targeting after building and unit testing.
+
+The propagation rules for additional environments such as any staging and production-type environments can't be specified via this mechanisms.
+These propagations tightly interact with the Probing & Attestation Phase and not infrequently require even manual or discretionary input.
+
+### Probing & Attestation Phase
+
+```yaml
+Register: n/a
+```
+
+The probing and attestation phase is highly situation specific.
+It cannot be adequately represented through Standardization and requires an entirely different control loop.
+
+Scenarios can include probing and attestation for the purpose of deploying a SaaS product into a specifit target infrastructure topology; or it could be targeting a managed on-site deployment.
```

Comment:
```suggestion
Scenarios can include probing and attestation for the purpose of deploying a SaaS product into a specific target infrastructure topology; or it could be targeting a managed on-site deployment.
```